streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -37,20 +38,37 @@
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Map;
-import java.util.Properties;
 import java.util.Set;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
 
 public class InternalTopicManager {
+    private final static String INTERRUPTED_ERROR_MESSAGE = "Thread got interrupted. This indicates a bug. " +
+        "Please report at https://issues.apache.org/jira/projects/KAFKA or dev-mailing list (https://kafka.apache.org/contact).";
+
+    private static final Map<String, String> REPARTITION_TOPIC_DEFAULT_OVERRIDES;
+    static {
+        final Map<String, String> tempTopicDefaultOverrides = new HashMap<>();
+        tempTopicDefaultOverrides.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_DELETE);
+        tempTopicDefaultOverrides.put(TopicConfig.SEGMENT_INDEX_BYTES_CONFIG, "52428800");     // 50 MB
+        tempTopicDefaultOverrides.put(TopicConfig.SEGMENT_BYTES_CONFIG, "52428800");           // 50 MB
+        tempTopicDefaultOverrides.put(TopicConfig.SEGMENT_MS_CONFIG, "600000");                // 10 min
```

Comment:
Do we really gain much with this setting?

Reply:
The segment size by default is 1GB, and purging would not take effects on the active segment. See https://issues.apache.org/jira/browse/KAFKA-6347 for details.

Reply:
My comment only targets `SEGMENT_MS_CONFIG` -- reducing segment size makes totally sense (sorry for not being clear in the first place).

Reply:
Sort of, though note that we may never get a completely empty repartition topic:

1) the segment rolling will only happen today upon appends, so if new data does not come in the current active segment will not be rolled even if `segment ms` has elapsed.
2) the goal of reducing this `segment ms` is that, for small traffic repartition topics, we could roll new segments and hence purge the old ones quicker before they hit size limit (50MB).

Reply:
Thanks for clarification.